tools/cli/domainCommands.go

```diff
@@ -447,13 +448,36 @@ func (d *domainCLIImpl) ListDomains(c *cli.Context) {
 	printAll := c.Bool(FlagAll)
 	printDeprecated := c.Bool(FlagDeprecated)
 	printFull := c.Bool(FlagPrintFullyDetail)
+	printJSON := c.Bool(FlagPrintJSON)
 
 	if printAll && printDeprecated {
 		ErrorAndExit(fmt.Sprintf("Cannot specify %s and %s flags at the same time.", FlagAll, FlagDeprecated), nil)
 	}
 
 	domains := d.getAllDomains(c)
 
+	if printJSON {
+		var filteredDomains []*types.DescribeDomainResponse
+		if printAll {
+			filteredDomains = domains
+		} else {
+			filteredDomains = make([]*types.DescribeDomainResponse, 0, len(domains))
```

Comment:
It seems that we are duplicating filtering logic, as it there is similar code for filtering when printing a table.

Could we make this filtering happen before, to be available for both rendering modes (table and json) ?